prov/sockets/src/sock_conn.c

```diff
@@ -376,7 +376,7 @@ int sock_conn_listen(struct sock_ep *ep)
 
 struct sock_conn *sock_ep_connect(struct sock_ep *ep, fi_addr_t index)
 {
-	int conn_fd, ret;
+	int conn_fd = 0, ret;
```

Comment:
It should not matter in either case. Even the initialization is not strictly required, since the conn_fd = socket() call is made always (except if conn!= SOCK_CM_CONN_IN_PROGRESS, in which case, it returns) . I added it because there is a check for conn_fd in error-exit path. In  future somebody modifies the code to take an error-exit before the conn_fd is created, it should not fail.

But, i can update it to -1, so that it is an invalid fd value.